src/utilities/plugin-utils.ts

```diff
@@ -109,38 +104,48 @@ export const validateEmbeddablePluginContextForWrappedEmbeddable =
 };
 
 // loads the learner plugin state into the firebase write-through cache
-export const loadLearnerPluginState = async (activities: Activity[], teacherEditionMode: boolean) => {
-  const plugins = findUsedPlugins(activities, teacherEditionMode);
-  return await Promise.all(plugins.map(async (plugin) => await getLearnerPluginState(plugin.id)));
+export const loadLearnerPluginState = async (activities: Activity[]) => {
+  const approvedScripts = findUsedApprovedScripts(activities);
+  // PJ 09/19/2021: This doesn't seem to make sense. Currently, the state is saved and restored per approved script.
+  // It should be saved and restored per plugin instance. It works with Glossary only because there's one glossary
+  // instance per activity. Fixing this would destroy students data, so I'm not doing this. But it should be handled
+  // if we ever add more plugins that save their state.
+  return await Promise.all(approvedScripts.map(async (approvedScriptInfo) => await getLearnerPluginState(approvedScriptInfo.id)));
 };
 
 export const initializePlugin = (context: IEmbeddablePluginContext) => {
   const { LARA, embeddable, embeddableContainer,
-          wrappedEmbeddable, wrappedEmbeddableContainer, sendCustomMessage, approvedScriptLabel } = context;
-  const usedPlugin = usedPlugins.find(p => p.plugin.approved_script_label === approvedScriptLabel);
-  if (!usedPlugin) return;
```

Comment:
As mentioned, this code existed as the AP initially only handled the teacher edition plugin. So we were intentionally filtering out the other plugins that we did not yet support. If we now have full plugin support, then it seems fine to remove.